### PR TITLE
Build overboard json

### PR DESCRIPTION
**Pull request by PietroCarrara**
Generate the JSON version of the overboard for the API

This commit also adds the information of which board a post belongs to the API

### CODE REVIEW

File: templates/themes/catalog/theme.php [hunk 1]
```diff
@@ -337,8 +337,8 @@ public function build($settings, $board_name) {
 
         private function buildThreadsQuery($board) {
             $sql  = "SELECT *, `id` AS `thread_id`, " .
-                "(SELECT COUNT(`id`) FROM ``posts_$board`` WHERE `thread` = `thread_id`) AS `reply_count`, " .
```
Comment (nonmakina) [inline templates/themes/catalog/theme.php:340]: What's the reasoning behind this change?

**Comment by nonmakina** [comment]
@PietroCarrara

**Comment by PietroCarrara** [comment]
Opened the PR a bit too early, but it's done now.

The only thing that I couldn't test was `/overboard/index.html`, since I don't know how to build that page, so checking if that still works is a good idea.

**Comment by nonmakina** [comment]
Sorry @PietroCarrara I don't quite understand how to use the review function on this site. 
I left a comment asking about the changes to the SQL queries. 
The question was what was the reasoning behind them.

**Comment by nonmakina** [comment]
Never mind. I checked the references and didn't find anything particularly worrying.

**Comment by nonmakina** [comment]
You should first have your branch up to date with upstream's `config`. The overboard is built as a theme. You need a mod account for it. I'll check it on dev. Just give me a sec.

**Review by nonmakina — APPROVED** [review]

**Pull request merged**